### PR TITLE
[HttpClient & Asp.Net Core] Update Readme.md

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #
Design discussion issue #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 1]
```diff
@@ -10,17 +10,15 @@ collect metrics and traces about incoming web requests. This instrumentation
 also collects traces from incoming gRPC requests using
 [Grpc.AspNetCore](https://www.nuget.org/packages/Grpc.AspNetCore).
 
-**Note: This component is based on the OpenTelemetry semantic conventions for
-[metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md)
+**Note: This component is based on the
+[v1.23](https://github.com/open-telemetry/semantic-conventions/tree/v1.23.0) of
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:14]: please add which attributes are added by default, and hint that more attributes may be added by using enrich

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 2]
```diff
@@ -10,17 +10,15 @@ collect metrics and traces about incoming web requests. This instrumentation
...
 and
-[traces](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md).
-These conventions are
-[Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/document-status.md),
-and hence, this package is a [pre-release](../../VERSIONING.md#pre-releases).
-Until a [stable
-version](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/telemetry-stability.md)
-is released, there can be breaking changes. You can track the progress from
-[milestones](https://github.com/open-telemetry/opentelemetry-dotnet/milestone/23).**
+[traces](https://github.com/open-telemetry/semantic-conventions/tree/v1.23.0/docs/http/http-spans.md).
```
Comment (TimothyMothra) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:18]: You may consider checking the *.cs files for additional links to the older conventions. We had them in several places when we got started on this.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/README.md]: why do we need older conventions?

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 3]
```diff
@@ -10,17 +10,46 @@ collect metrics and traces about incoming web requests. This instrumentation
 also collects traces from incoming gRPC requests using
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:10]: this needs to called out as experimental

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 4]
```diff
@@ -10,17 +10,46 @@ collect metrics and traces about incoming web requests. This instrumentation
...
-and
-[traces](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md).
-These conventions are
-[Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/document-status.md),
-and hence, this package is a [pre-release](../../VERSIONING.md#pre-releases).
-Until a [stable
-version](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/telemetry-stability.md)
-is released, there can be breaking changes. You can track the progress from
-[milestones](https://github.com/open-telemetry/opentelemetry-dotnet/milestone/23).**
+**Note: This component is based on the
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:13]:
```suggestion
This component is based on the
```

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 5]
```diff
@@ -10,17 +10,46 @@ collect metrics and traces about incoming web requests. This instrumentation
...
+[v1.23](https://github.com/open-telemetry/semantic-conventions/tree/v1.23.0/docs/http)
+of http semantic conventions.
+
+Following list of attributes are added by default on activity. See
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:17]: split it under tracing, metrics heading separately to make it easy to follow.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/README.md]: I moved it under traces and metrics below. Please check if that will be more clear.

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md (hunk 3 above)
Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/README.md:10]: done

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md (hunk 2 above)
Comment (TimothyMothra) [reply src/OpenTelemetry.Instrumentation.AspNetCore/README.md]: Sorry, I misspoke. While working on this the last few months we referenced the specification that was available at the time. I meant to say, we should review the codebase to confirm that all the older links were replaced with the current 1.23

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/README.md]: I think we did that already. Will do the clean up if needed separately.

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 6]
```diff
@@ -254,6 +285,31 @@ The time ends when:
 * All response data has been sent.
 * The context data structures for the request are being disposed.
 
+## Experimental support for gRPC requests
+
+gRPC instrumentation can be enabled by setting
+`OTEL_DOTNET_EXPERIMENTAL_ASPNETCORE_ENABLE_GRPC_INSTRUMENTATION` flag to
+`True`. The flag can be set as an environment variable or via IConfiguration as
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:292]: nit: Use the same casing throughout the doc. The code snippet below uses `true`.

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 7]
```diff
@@ -66,6 +61,26 @@ public void ConfigureServices(IServiceCollection services)
 }
 ```
 
+Following list of attributes are added by default on activity. See
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:64]: A lot of these attributes are only added when they are available. For example, `error.type`, `user_agent.original` etc. 

```suggestion
Following list of attributes are added by default on activity when available. See
```

File: src/OpenTelemetry.Instrumentation.Http/README.md [hunk 8]
```diff
@@ -65,6 +57,22 @@ public class Program
 }
 ```
 
+Following list of attributes are added by default on activity. See
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Http/README.md:60]: Same comment as https://github.com/open-telemetry/opentelemetry-dotnet/pull/5168/files#r1426122165

**Review by utpilla — APPROVED** [review]

**Comment by utpilla** [comment]
@vishweshbankwar Left some suggestions that could be addressed in a follow-up PR.

**Pull request merged**